components/brave_ads/core/internal/ads/serving/targeting/user_model_builder.cc

```diff
@@ -34,7 +41,28 @@ UserModelInfo BuildUserModel() {
     user_model.purchase_intent_segments = purchase_intent_model.GetSegments();
   }
 
-  return user_model;
+  if (features::IsTextEmbeddingEnabled()) {
+    GetTextEmbeddingHtmlEventsFromDatabase(base::BindOnce(
+        &OnGetTextEmbeddingHtmlEvents, user_model, std::move(callback)));
+  } else {
+    std::move(callback).Run(user_model);
+  }
+}
+
+void OnGetTextEmbeddingHtmlEvents(
```

Comment:
can we please make this a non-member function in an anonymous namespace and remove from the interface